frame/metadata/src/lib.rs

```diff
@@ -407,6 +409,9 @@ pub struct ModuleMetadata {
 	pub event: ODFnA<EventMetadata>,
 	pub constants: DFnA<ModuleConstantMetadata>,
 	pub errors: DFnA<ErrorMetadata>,
+	/// Define the fix index of the module, this index will be used for the encoding of module
+	/// event, call and origin variants if it has some.
+	pub fixed_index: Option<u8>,
```

Comment:
what you mean is that user will have to write for every module `= $n;` isn't it ?

Reply:
No, polkadot-js already always assumed some implicit index which is derived from the order of the data in the metadata. Instead of making this implicit, we should make it explicit.

Not sure if we should force the user to give the index for every module. Maybe we should do the following:

- If no index is given, we take the implicit location of the module in the macro call.
- If some module before has a fixed index, every module following it will be using `n + 1` as index. Example
```
Module = 10;
Module2 (will get 11);
Module3 (will get 12);
```

WDYT?

Reply:
Or we force an index for every module, in the moment the user sets any index.

Reply:
I'm ok to come up with a strategy to automatically fix implicit index.

Just to be sure, that means that we agree to have this breaking change:

If we do `If no index is given, we take the implicit location of the module in the macro call.` then the following runtime
```
..
system: ...,
module1: Module1::{},
module2: Module2::{Origin},
```
Will see the encoding of the runtime origin changing, module2 will be encoded at index 2 instead of 1.
(and same for call).

Reply:
> Will see the encoding of the runtime origin changing, module2 will be encoded at index 2 instead of 1.
(and same for call).

Yeah, exactly!

Reply:
FInally I implemented:

>      Module = 10;
Module2 (will get 11);
Module3 (will get 12);

Because It is same as how it works for enum in rust. And I think implicit is somewhat handy for quick test definition.